### PR TITLE
Several fixes and improvements, new performance plot added [ci skip]

### PR DESCRIPTION
**Pull request by reimersa**

### CODE REVIEW

File: DNNTools/PreprocessInputsBase.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def GetInputs(self,format='csv'):
         self.df = pd.concat(inputs, ignore_index=True)
         del inputs
         self.df.rename(lambda x: x[0] if isinstance(x, tuple) else x , axis='columns', inplace=True)
+        print self.df[['category']]
```
Comment (anmalara) [inline DNNTools/PreprocessInputsBase.py:42]: can this go?

File: DNNTools/PreprocessInputsBase.py [hunk 2]
```diff
@@ -81,9 +84,9 @@ def SaveBase(self, format='csv'):
         frac = float_to_str(self.runonfraction)
         outdir = os.path.join(self.outdir, classes_to_str(self.DefineClasses()))
         for mode in ['train', 'val', 'test']:
-            SavePandas(self.inputs[mode],  os.path.join(outdir, 'inputs_%s_%s.%s'  %(mode,frac,format)))
-            SavePandas(self.weights[mode], os.path.join(outdir, 'weights_%s_%s.%s' %(mode,frac,format)))
-            SavePandas(self.labels[mode],  os.path.join(outdir, 'labels_%s_%s.%s'  %(mode,frac,format)))
+            SavePandas(self.inputs[mode].astype('float32'),  os.path.join(outdir, 'inputs_%s_%s.%s'  %(mode,frac,format)))
+            SavePandas(self.weights[mode].astype('float32'), os.path.join(outdir, 'weights_%s_%s.%s' %(mode,frac,format)))
+            SavePandas(self.labels[mode].astype('int8'),  os.path.join(outdir, 'labels_%s_%s.%s'  %(mode,frac,format)))
```
Comment (anmalara) [inline DNNTools/PreprocessInputsBase.py:89]: stored as int8, but loaded as float32?

File: DNNTools/PlottingFunctions.py [hunk 3]
```diff
@@ -35,28 +35,30 @@ def plot_losses(history, mode='loss', name='history', min_epoch=0, dynamic=True,
     canv.Close()
 
 
-def plot_rocs(rocs={}, name='ROCs', x_title='Signal efficiency', y_title='Background efficiency', logy=True, writeExtraText = True, extraText  = 'Simulation', extraText2 = 'Work in progress', lumi_text=''):
+def plot_graphs(graphs={}, name='ROCs', x_title='Signal efficiency', y_title='Background efficiency', x_range=(-0.1, 1.1), y_range=(1e-04, 1.2), logy=True, writeExtraText = True, extraText  = 'Simulation', extraText2 = 'Work in progress', lumi_text=''):
```
Comment (anmalara) [inline DNNTools/PlottingFunctions.py:38]: why the change in name? because it's more general?

File: DNNTools/PlotterBase.py [hunk 4]
```diff
@@ -66,57 +70,78 @@ def PlotSingleVariable(self, dfs, weights, labels, style, variable_name, outdir,
...
         for i in sorted(labels[labels.columns[0]].unique()):
             g_roc = list_to_tgraph(TruePositiveRates[i], FalsePositiveRates[i])
-            g_pur = list_to_tgraph(TruePositiveRates[i], SignalPuritys[i])
-            g_pur_vs_score = list_to_tgraph(Thresholds[i], SignalPuritys[i])
+            g_pur = list_to_tgraph(TruePositiveRates[i][1:], SignalPuritys[i][1:])
+            g_pur_vs_score = list_to_tgraph(Thresholds[i][1:], SignalPuritys[i][1:])
+            sumweights_sig = weights[(labels[labels.columns[0]] == i)][weights.columns[0]].sum()
+            sumweights_backgrounds = weights[~(labels[labels.columns[0]] == i)][weights.columns[0]].sum()
+            pur_inclusive = sumweights_sig / (sumweights_sig + sumweights_backgrounds)
+            g_pur_improvements_vs_score = list_to_tgraph(Thresholds[i][1:], SignalPuritys[i][1:] / pur_inclusive)
```
Comment (anmalara) [inline DNNTools/PlotterBase.py:87]: what is this?

File: DNNTools/PreprocessInputsBase.py (hunk 1 above)
Comment (reimersa) [reply DNNTools/PreprocessInputsBase.py:42]: yes

File: DNNTools/PreprocessInputsBase.py [hunk 5]
```diff
@@ -55,6 +56,8 @@ def SampleEvents(self, fraction=1.0, weights=None, random_state=123456):
 
     def Split(self, ratios={'train':0.8, 'validation':0.1, 'test':0.1}):
         weights = pd.DataFrame(self.df.loc[:,self.colname_weights], columns=[self.colname_weights])
+        print self.DefineClasses()
+        print self.df[[self.colname_category]]
```
[the remark this answers is not in the document]
Comment (reimersa) [reply DNNTools/PreprocessInputsBase.py:60]: yes

File: DNNTools/PreprocessInputsBase.py (hunk 2 above)
Comment (reimersa) [reply DNNTools/PreprocessInputsBase.py:89]: yes

File: DNNTools/PlottingFunctions.py (hunk 3 above)
Comment (reimersa) [reply DNNTools/PlottingFunctions.py:38]: yes

**Comment by reimersa** [comment]
fixed printouts and shortened normalization, ready to merge?

File: DNNTools/PlotterBase.py [hunk 6]
```diff
@@ -55,9 +55,13 @@ def PlotSingleVariable(self, dfs, weights, labels, style, variable_name, outdir,
                 weights_thisclass = weights[name][mask]
                 weights_thisclass = weights_thisclass[weights_thisclass.columns[0]]
                 df_thisclass = df[mask][variable_name]
+                if normalize_all:
+                    sumweights_thisclass = weights_thisclass.sum()
+                    weights_thisclass *= 1./sumweights_thisclass
```
[the remark this answers is not in the document]
Comment (anmalara) [reply DNNTools/PlotterBase.py]: nicer in python and you avoid storing objects in memory

File: DNNTools/PlotterBase.py (hunk 4 above)
Comment (anmalara) [reply DNNTools/PlotterBase.py:86]: and what does it represent?

File: DNNTools/PreprocessInputsBase.py (hunk 2 above)
Comment (reimersa) [reply DNNTools/PreprocessInputsBase.py:86]: saves space

File: DNNTools/PlotterBase.py (hunk 4 above)
Comment (reimersa) [reply DNNTools/PlotterBase.py:86]: the improvement/deterioration of s/(s+b) for a given cut x relative to the inclusive s/(s+b) [i.e. the factor by which the s/(s+b) changes when cutting at a value x compared to the inclusive s/(s+b) before the cut]
